ammico/summary.py

```diff
@@ -169,99 +236,364 @@ def load_vqa_model(self):
         )
         return summary_vqa_model, summary_vqa_vis_processors, summary_vqa_txt_processors
 
-    def analyse_image(self):
+    def analyse_image(
+        self,
+        analysis_type: Optional[str] = None,
+        subdict: dict = {},
+        list_of_questions: Optional[list[str]] = None,
+        consequential_questions: bool = False,
+    ):
         """
         Analyse image with blip_caption model.
 
         Args:
+            analysis_type (str): type of the analysis.
+            subdict (dict): dictionary with analising pictures.
+            list_of_questions (list[str]): list of questions.
+            consequential_questions (bool): whether to ask consequential questions. Works only for new BLIP2 models.
 
         Returns:
             self.subdict (dict): dictionary with analysis results.
         """
+        if analysis_type is not None:
+            self.analysis_type = analysis_type
+        if bool(subdict):
+            self.subdict = subdict
+        if list_of_questions is not None:
+            self.list_of_questions = list_of_questions
         if self.analysis_type == "summary_and_questions":
-            self.analyse_summary()
-            self.analyse_questions(self.list_of_questions)
+            self.analyse_summary(nondeterministic_summaries=True)
+            self.analyse_questions(self.list_of_questions, consequential_questions)
         elif self.analysis_type == "summary":
-            self.analyse_summary()
+            self.analyse_summary(nondeterministic_summaries=True)
         elif self.analysis_type == "questions":
-            self.analyse_questions(self.list_of_questions)
-
+            self.analyse_questions(self.list_of_questions, consequential_questions)
+        else:
+            raise ValueError(
+                "analysis_type must be one of {}".format(self.allowed_analysis_types)
+            )
         return self.subdict
 
-    def analyse_summary(self):
+    def analyse_summary(self, nondeterministic_summaries: bool = True):
         """
         Create 1 constant and 3 non deterministic captions for image.
 
         Args:
+            nondeterministic_summaries (bool): whether to create 3 non deterministic captions.
 
         Returns:
             self.subdict (dict): dictionary with analysis results.
         """
-
+        if self.model_type in self.allowed_model_types:
```

Comment:
so there is some repetition here with `__init__`

Reply:
I use this for convenience, so that with both new and old models I can continue to write the same code.